packages/twenty-server/src/workspace/messaging/repositories/message/message.service.ts

```diff
@@ -203,12 +203,30 @@ export class MessageService {
         workspaceId,
       );
 
-    if (isContactAutoCreationEnabled) {
+    const isOutgoing = connectedAccount.handle === message.fromHandle;
```

Comment:
You already do this above for messageDirection, I think you can use messageDirection directly https://github.com/twentyhq/twenty/pull/3995/files#diff-61fadec3f31d9c5f3f7a3924db219d9397b6a7568bcd20bb408b4014ac028fc5R181